File: 3820-number-of-unique-xor-triplets-ii/number-of-unique-xor-triplets-ii.java
public class Solution {
    public int uniqueXorTriplets(int[] nums) {
        int m=0;
        for (int num : nums)
            m=Math.max(m, num);
        int T = 1;
        while (T <= m)
            T <<= 1;
        boolean[] s1 = new boolean[T];
        boolean[] s2 = new boolean[T];
        for (int i = 0; i < nums.length; i++)
            for (int j = i; j < nums.length; j++)
                s1[nums[i] ^ nums[j]] = true;
        for (int i = 0; i < T; i++)
            if (s1[i]) 
                for (int num : nums) 
                    s2[i ^ num] = true;
        int c= 0;
        for (int i = 0; i < T; i++) 
            if (s2[i]) 
                c++;        
        return c;
    }
}

File: 3820-number-of-unique-xor-triplets-ii/number-of-unique-xor-triplets-ii.c
int uniqueXorTriplets(int* nums, int numsSize) {
    int m= 0;
    for (int i = 0; i < numsSize; i++)
        if (nums[i] > m) 
            m= nums[i];
        
    int t = 1;
    while (t <= m)
        t= t * 2;
    
    bool* s1 = (bool*)calloc(t, sizeof(bool));
    bool* s2 = (bool*)calloc(t, sizeof(bool));
    for (int i = 0; i < numsSize; i++)
        for (int j = i; j < numsSize; j++)
            s1[nums[i] ^ nums[j]] = true;
    
    for (int i = 0; i < t; i++) 
        if (s1[i]) 
            for (int j = 0; j < numsSize; j++) 
                s2[i ^ nums[j]] = true;
         
    int count = 0;
    for (int i = 0; i < t; i++) 
        if (s2[i]) 
            count++;
    free(s1);
    free(s2);
    
    return count;
}